color-bolts/src/main/java/org/dmfs/android/bolts/color/dynamic/lightness/Toned.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.bolts.color.dynamic.lightness;
+
+import android.support.v4.graphics.ColorUtils;
+
+import org.dmfs.android.bolts.color.Color;
+import org.dmfs.jems.function.Function;
+
+
+/**
+ * {@link Color} decorator to adjust lightness by applying a function to its HSL-Lightness value.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class Toned implements Color
+{
+    private final Color mOriginal;
+    private final Function<Float, Float> mLightnessAdjustingFunction;
+
+
+    /**
+     * Creates a new instance with adjusting the original color's lightness.
+     *
+     * @param lightnessAdjustingFunction
+     *         applied to HSV-Value(Lightness), but the new value will always be clipped to the valid [0,1] range regardless of the result of the function
+     * @param original
+     *         the color to change
+     */
+    public Toned(Function<Float, Float> lightnessAdjustingFunction, Color original)
+    {
+        mLightnessAdjustingFunction = lightnessAdjustingFunction;
+        mOriginal = original;
+    }
+
+
+    @Override
+    public int argb()
+    {
+        float[] hsl = new float[3];
+        ColorUtils.colorToHSL(mOriginal.argb(), hsl);
+
+        hsl[2] = Math.min(Math.max(mLightnessAdjustingFunction.value(hsl[2]), 0), 1);
+
+        return ColorUtils.HSLToColor(hsl);
```

Comment:
this still doesn't set the alpha of the result. The alpha value should remain the same.

Reply:
I've updated it to keep the alpha.